scenes/actors/Player/RegionMovement.gd

```diff
@@ -0,0 +1,62 @@
+tool
+extends State
+
+var player : Player
+
+#
+# FUNCTIONS TO INHERIT IN YOUR STATES
+#
+
+onready var gravity = ProjectSettings.get_setting("physics/2d/default_gravity")
+
+# This function is called when the state enters
+# XSM enters the root first, the the children
+func _on_enter(_args) -> void:
+	player = target as Player
+	
+	if player.is_on_floor():
+		var _st = change_state("OnGround")
+	else:
+		var _st = change_state("InAir")
```

Comment:
Do these local variables need to be kept here or can we just change_state directly

Reply:
you can change it directly, but it throws a warning so my OCD got the better of me